### PR TITLE
Document YIELD and WHERE for SHOW INDEXES

### PR DESCRIPTION
**Pull request by Hunterness**
Also deprecate BRIEF and VERBOSE keywords.

Depends on https://github.com/neo-technology/neo4j/pull/8855 (merged)

### CODE REVIEW

File: cypher/cypher-prettifier/src/test/scala/org/neo4j/cypher/docgen/tooling/PrettifierTest.scala [hunk 1]
```diff
@@ -68,8 +68,11 @@ class PrettifierTest extends Suite
     actual("drop index name if exists") should equal(expected("DROP INDEX name IF EXISTS"))
   }
 
-  test("should not break SHOW INDEXES") {
+  test("may break SHOW INDEXES") {
```
Comment (OliviaYtterbrink) [inline cypher/cypher-prettifier/src/test/scala/org/neo4j/cypher/docgen/tooling/PrettifierTest.scala]: I think this should be split in two tests, one for breaks and one for the old behaviour
and shouldn't `show indexes brief` be here as well?

**Review by OliviaYtterbrink — CHANGES_REQUESTED** [review]

File: cypher/cypher-prettifier/src/test/scala/org/neo4j/cypher/docgen/tooling/PrettifierTest.scala (hunk 1 above)
Comment (Hunterness) [reply cypher/cypher-prettifier/src/test/scala/org/neo4j/cypher/docgen/tooling/PrettifierTest.scala]: what about line 76?

Comment (Hunterness) [reply cypher/cypher-prettifier/src/test/scala/org/neo4j/cypher/docgen/tooling/PrettifierTest.scala]: but yes, I should probably split it

**Review by OliviaYtterbrink — APPROVED** [review]

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc [hunk 2]
```diff
@@ -103,7 +103,10 @@ With `IF EXISTS`, no error is thrown and nothing happens should the index not ex
 
 | [source, cypher, role=noplay]
 ----
-SHOW [ALL\|BTREE] INDEX[ES] [BRIEF\|VERBOSE [OUTPUT]]
+SHOW [ALL\|BTREE] INDEX[ES]
+    [YIELD { * \| field[, ...] } [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
+    [WHERE expression]
+    [RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
```
Comment (Hunterness) [inline cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc:109]: Update to match https://trello.com/c/XIambksk/2372-update-docs-to-have-valid-where-yield-return-in-show-syntax-descriptions

**Review by martin-neotech — APPROVED** [review]
Thanks

**Pull request merged**